.pre-commit-config.yaml

```diff
@@ -52,11 +57,24 @@ repos:
   hooks:
     - id: ruff
       exclude: tests/
-      args: [--line-length=120]
+      args: [--line-length=120, --fix]
```

Comment:
I just found some qwirks with line length and pre commit hooks. Should we keep 120 chars per line or 80? I think 100 is the python default currently, isn't it?

Reply:
According to [PEP8](https://peps.python.org/pep-0008/#:~:text=The%20Python%20standard%20library%20is,inside%20parentheses%2C%20brackets%20and%20braces.), even 79. Personally I think 100 is fine. :)

Reply:
And I just found out, that you have to run the pre-commit a dozent times until everything is checked.

I would like to cherry pick your change here to a branch I already made, which will introduce all pre commit changes, ok?

Reply:
Sure. I can run a pre-commit --all-files and work through the feedback from the guidelines. Should bring up the standards and prevents cherry picking. :)

Reply:
I still have the problem, that vscode formatter is using different formatting rules than the pre-commit hooks.

Reply:
Do you have a screenshot and know which Linter is causing this issue?

Reply:
we currently use black. So when in vscode black does some formatting, and when I run `pre-commit run -a` black is reformatting again 😄

Reply:
I see. Not sure where that is located, but it would help if one Black formatter is triggered. :D

Reply:
easiest would be to deactivate formatting in vscode, so everybody is forced to use, what the pre-commit hook "says". But I personally like, that everything is nicely formatted when I save! The wait time until committing is a bit uncommon for me!

Reply:
Then we should update this in the pyproject.toml rules. Both VScode and pre-commit will use the shared configuration.